haystack/components/converters/azure.py

```diff
@@ -51,16 +51,10 @@ def __init__(self, endpoint: str, api_key: Optional[str] = None, model_id: str =
         """
         azure_import.check()
 
-        if api_key is None:
-            try:
-                api_key = os.environ["AZURE_AI_API_KEY"]
-            except KeyError as e:
-                raise ValueError(
-                    "AzureOCRDocumentConverter expects an Azure Credential key. "
-                    "Set the AZURE_AI_API_KEY environment variable (recommended) or pass it explicitly."
-                ) from e
-
-        self.api_key = api_key
+        api_key = get_api_key_from_param_or_env(
+            component_name=self.__class__.__name__, param=api_key, environment_variable="AZURE_AI_API_KEY"
+        )
```

Comment:
Personal opinion, but this is pretty ugly compared to
```py
api_key = api_key or os.environ.get("AZURE_AI_API_KEY")
if not api_key:
        msg = (
            f"AzureOCRDocumentConverter expects an API key. "
            f"Set the AZURE_AI_API_KEY environment variable (recommended) or pass it explicitly."
        )
        raise ValueError(msg)
```
Very good job uniforming the logic across the codebase, but I don't think we need a wrapper...

Reply:
Okay. So you're suggesting to standardize the logic but not to introduce the utility function?

Reply:
Correct. If you really, really want to enforce it, in my opinion a decorator would work much better:

```py
@ensure_api_key(fallback="AZURE_AI_API_KEY")
def __init__(self, endpoint: str, api_key: Optional[str] = None, model_id: str = "prebuilt-read"):
    ...
```
(the implementation should be fairly easy)

Reply:
I decided to remove the utility function and be always explicit.
(I don't introduce the decorator)